contracts/Smt.metasol

```diff
@@ -17,6 +17,7 @@ library SMT{{hashname}} {
     function initialize(MerkleTree storage tree, uint8 _depth) internal {
         require(tree.depth == 0, "Cannot initialize twice");
         require(_depth > 0, "Depth must be non-zero");
+        require(_depth < 255, "Depth must be < 255");
```

Comment:
why not `<=`?

Reply:
OK, so basically our tree is actually restricted to depth 254? so `max(totalElements)` (which is now commented) is 2^254.

Reply:
>why not <=?

Because it's doesn't sense. Any uint8 <= 255 by default

Reply:
>OK, so basically our tree is actually restricted to depth 254? so max(totalElements) (which is now commented) is 2^254.

Oh i will fix it

Reply:
oh I see, I wanted to ask why we don't specify anywhere _max depth_ but I see now that the depth type is uint 8.
So more broadly, wouldn't we want to allow larger (or smaller?) trees in the future?

Reply:
You can have trees with depth ∈ {1, 2 ... 254 }

So you can have depth less than 254 but not more

I think 2^254 it's enough elements:

28948022309329048855892746252171976963317496166410141009864396001978282409984

Reply:
mmm.... yeah, I think my math got broken : )
so I think that the issue that was found, while indeed a bug, is a bit meaningless. One will go on forever to initialize the 254-depth tree anyway.

Regarding the smaller tree sizes, yes I understand that. I was basically asking about a situation where we want to _limit_ the tree's depth into something smaller. For example, we may only want to allow 1024 participants.

Reply:
Right now we can limit in sc where we use this smt library. Also anyway we did in our sc. We have limit 10.